java/cdk/src/main/java/sleeper/cdk/stack/bulkimport/EmrServerlessBulkImportStack.java

```diff
@@ -0,0 +1,245 @@
+        createEmrServerlessApplication(instanceProperties);
+
+        IRole emrRole = createEmrServerlessRole(scope, instanceProperties);
+
+        CommonEmrBulkImportHelper commonHelper = new CommonEmrBulkImportHelper(this,
+                "EMRServerless", instanceProperties, statusStoreResources);
+        Queue bulkImportJobQueue = commonHelper.createJobQueue(
+                BULK_IMPORT_EMR_SERVERLESS_JOB_QUEUE_URL, errorsTopicStack.getTopic());
+        IFunction jobStarter = commonHelper.createJobStarterFunction("EMRServerless",
+                bulkImportJobQueue, jars, importBucketStack.getImportBucket(), List.of(emrRole));
+        stateStoreStacks.forEach(sss -> sss.grantReadPartitionMetadata(jobStarter));
+        configureJobStarterFunction(instanceProperties, jobStarter);
+        Utils.addStackTagIfSet(this, instanceProperties);
+    }
+
+    private static void configureJobStarterFunction(InstanceProperties instanceProperties,
+            IFunction bulkImportJobStarter) {
+        Map<String, Map<String, String>> conditions = new HashMap<>();
+        Map<String, String> tagKeyCondition = new HashMap<>();
+        instanceProperties.getTags().forEach(
+                (key, value) -> tagKeyCondition.put("elasticmapreduce:RequestTag/" + key, value));
+
+        conditions.put("StringEquals", tagKeyCondition);
+
+        bulkImportJobStarter.addToRolePolicy(PolicyStatement.Builder.create()
+                .actions(List.of("elasticmapreduce:RunJobFlow")).effect(Effect.ALLOW)
+                .resources(List.of("*")).conditions(conditions).build());
+    }
+
+    public void createEmrServerlessApplication(InstanceProperties instanceProperties) {
+        String instanceId = instanceProperties.get(ID);
+        String region = instanceProperties.get(REGION);
+        String accountId = instanceProperties.get(ACCOUNT);
+        String repo = instanceProperties.get(BULK_IMPORT_EMR_SERVERLESS_CUSTOM_IMAGE_REPO);
+        String version = instanceProperties.get(VERSION);
+        String uri = accountId + ".dkr.ecr." + region + ".amazonaws.com/" + repo + ":" + version;
+
+        CfnApplicationProps props = CfnApplicationProps.builder()
+                .name(String.join("-", "sleeper", instanceId, "emr", "serverless"))
+                .releaseLabel(instanceProperties.get(BULK_IMPORT_EMR_SERVERLESS_RELEASE))
+                .architecture(instanceProperties.get(BULK_IMPORT_EMR_SERVERLESS_ARCHITECTURE))
+                .type("Spark")
+                .imageConfiguration(ImageConfigurationInputProperty.builder().imageUri(uri).build())
+                .networkConfiguration(NetworkConfigurationProperty.builder()
+                        .subnetIds(List.of(getSubnet(instanceProperties)))
+                        .securityGroupIds(List.of(createSecurityGroup(instanceProperties))).build())
+                .tags(List.of(new CfnTag.Builder().key("DeploymentStack")
+                        .value("EmrServerlessBulkImport").build()))
```

Comment:
Is this necessary? This tag is already added to the stack as a whole. The name of the tag for this is also configurable, see Utils.addStackTagIfSet.

Reply:
I have removed this as its not needed